.gitlab/container_build/docker_linux.yml

```diff
@@ -18,6 +18,22 @@
     - docker-squash ${TARGET_TAG}-unsquashed -t ${TARGET_TAG}
     - test "$TEST_IMG" && docker run -v `pwd`/$BUILD_CONTEXT:/tmp/build ${TARGET_TAG} python /tmp/build/test_image_contents.py
     - docker push $TARGET_TAG
+    ## Mirror container to single-machine-performance to serve as 'baseline' and
+    ## 'comparison' targets.
+    # Setup AWS credentials for single-machine-performance AWS account
+    - SMP_ACCOUNT_ID=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.single-machine-performance-account-id --with-decryption --query "Parameter.Value" --out text)
```

Comment:
Do you really need **all** images? (e.g. including dev branches, etc.)?
If so, I think the cleanest way is probably to use ECR cross-account replication:
https://docs.aws.amazon.com/AmazonECR/latest/userguide/registry-settings-configure.html

If this option has been ruled out for some reasons, I believe this setup is a bit dangerous is the sense that if the job is extended, it will silently use SMP AWS credentials instead of ddbuild runner credentials, which will probably come at a surprise.
I think it's better to simply allow runner IAM user to push to SMP ECR, see:
https://aws.amazon.com/premiumsupport/knowledge-center/secondary-account-access-ecr/

Reply:
> Do you really need **all** images? (e.g. including dev branches, etc.)? If so, I think the cleanest way is probably to use ECR cross-account replication: https://docs.aws.amazon.com/AmazonECR/latest/userguide/registry-settings-configure.html

For now, no, we really only want Agent 7 x86. I previously had a job that duplicated only that build but pushed to our ECR. I could re-introduce that if that's preferable. From what I understand of the cross-account replication rules it looks like this would need to be set up in the Agent CI account? If so, we'd need guidance on where to introduce changes in your terraform to set that up.
 
> If this option has been ruled out for some reasons, I think it's better to simply allow runner IAM user to push to SMP ECR, see: https://aws.amazon.com/premiumsupport/knowledge-center/secondary-account-access-ecr/

Sure, I'll contact you out of band to discuss.